src/Tracing/DynamicSamplingContext.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\Tracing;
+
+use Sentry\State\HubInterface;
+use Sentry\State\Scope;
+
+/**
+ * This class represents the Dynamic Sampling Context (dsc).
+ *
+ * @see https://develop.sentry.dev/sdk/performance/dynamic-sampling-context/
+ */
+final class DynamicSamplingContext
+{
+    public const SENTRY_ENTRY_PREFIX = 'sentry-';
+
+    /**
+     * The dsc entries.
+     *
+     * @var array<string, string>
+     */
+    private $entries;
+
+    /**
+     * Indicates that the dsc is frozen and cannot be mutated.
+     *
+     * @var bool
+     */
+    private $frozen = false;
```

Comment:
Why is this `false`?

Reply:
Also, for boolean variable names, it is always better to prefix them with `is`, `has` or something similar, where possible. This could be easier to read as `isFrozen`.

Reply:
By default, a DSC is mutable until you add values to it. This refelts that.